t/t0100-modprobe.t

```diff
@@ -859,7 +859,7 @@ test_expect_success 'modprobe: fails if module load fails' '
 	flux python <<-EOF >all_modules &&
 	import flux
 	from flux.modprobe import ModuleList
-	modules = [x for x in ModuleList(flux.Flux()) if x != "connector-local"]
+	modules = [x for x in ModuleList(flux.Flux()) if x != "connector-local" and x != "groups"]
```

Comment:
could also do
```python
if x not in ("connector-local", "groups")
```

If there is some way to determine that a module auto-loaded, we could extend modprobe to handle these things better.

Reply:
A thought is to 1) take your advice a bit late and eliminate non-auto-loaded builtin modules, and 2) use the module path of `"builtin"` as a key to omit those modules from `ModuleList` output.  The same could be done by using a special path for the auto-loaded built-ins but sheesh, more complexity for a feature that isn't buying us much.

Reply:
> 1) take your advice a bit late and eliminate non-auto-loaded builtin modules,

Hm, I didn't mean to suggest to eliminate the other builtin modules.

I think an `autoload` boolean flag in the `module.list` RPC might be sufficient, then as you suggest above `ModuleList` could just ignore those.

Another approach could be just to have the modprobe `ModuleList` ignore modules that aren't within its configuration. In retrospect this might make the most sense since `ModuleList` is a modprobe helper...